programs/server/Server.cpp

```diff
@@ -562,6 +560,37 @@ int Server::main(const std::vector<std::string> & /*args*/)
 
     StatusFile status{path + "status", StatusFile::write_full_info};
 
+
+    /// Write a uuid file containing a unique uuid if the file doesn't already exist during server start.
+    {
+        fs::path server_uuid_file = fs::path(path) / "uuid";
+
+        if (!fs::exists(server_uuid_file))
+        {
+            try
+            {
+                /// Note: Poco::UUIDGenerator().createRandom() uses /dev/random and can be expensive. But since
+                /// it's only going to be generated once (i.e if the uuid file doesn't exist), it's probably fine.
+                auto uuid_str = Poco::UUIDGenerator().createRandom().toString();
+                WriteBufferFromFile out(server_uuid_file.string());
+                out.write(uuid_str.data(), uuid_str.size());
+                out.sync();
+                out.finalize();
+            }
+            catch (...)
+            {
+                std::string message
```

Comment:
Formatting might look odd - but it's actually legit and made by editor. Can't seem to change it to anything better.